build_runner/lib/src/entrypoint/options.dart

```diff
@@ -399,3 +419,36 @@ Set<BuildDirectory> _parseBuildDirs(ArgResults argResults) {
   }
   return result;
 }
+
+/// Returns build filters parsed from [buildFilterOption] arguments.
+///
+/// These support `package:` uri syntax as well as regular path syntax,
+/// with glob support.
+List<BuildFilter> _parseBuildFilters(
+    ArgResults argResults, String rootPackage) {
+  var filterArgs = argResults[buildFilterOption] as List<String>;
+  if (filterArgs?.isNotEmpty != true) return null;
+  var filters = <BuildFilter>[];
+  for (var arg in filterArgs) {
```

Comment:
What if we add a `BuildFilter.fromArg(String rootPackage, String arg)` constructor, then we could do something like this here:

```dart
try {
  return [
    for(var arg in filterArgs)
      BuildFilter.forArg(rootPackage, arg),
  ];
} on FormatException catch(e) {
  throw ArgumentError.value(...);
}
```

Reply:
Sure, done